pandas/core/common.py

```diff
@@ -216,13 +215,7 @@ def try_sort(iterable):
 
 
 def dict_keys_to_ordered_list(mapping):
-    # when pandas drops support for Python < 3.6, this function
-    # can be replaced by a simple list(mapping.keys())
-    if PY36 or isinstance(mapping, OrderedDict):
-        keys = list(mapping.keys())
-    else:
-        keys = try_sort(mapping)
-    return keys
+    return list(mapping.keys())
```

Comment:
It looks like there are only three references to this aside from the definition here:
```
./pandas/core/reshape/concat.py:296:dict_keys_to_ordered_list
./pandas/core/internals/construction.py:238:dict_keys_to_ordered_list
./pandas/core/common.py:218:dict_keys_to_ordered_list
./pandas/tests/reshape/test_concat.py:1267:dict_keys_to_ordered_list
```
It doesn't look like any of these instances actually rely on this being a list, and the dictview `mapping.keys()` should be fine.  So I think we can just remove this function and replace the three calls with `whatever.keys()`.  Fine to make this a follow-up.